.github/workflows/lint.yaml

```diff
@@ -0,0 +1,22 @@
+---
+name: "Lint"
+on:  # yamllint disable-line rule:truthy
+  push:
+    branches:
+      - "main"
+  pull_request:
+    branches: ["*"]
+jobs:
+  lint:
+    # Uses pre-commit to run all checks
+    runs-on: "ubuntu-latest"
+    steps:
+      - uses: "actions/checkout@v4"
+      # needed for pre-commit
+      - uses: "actions/setup-python@v5"
+        with:
+          python-version: "3.11"
+      - uses: "actions/setup-dotnet@v4"
+        with:
+          dotnet-version: '8.x'
+      - uses: "pre-commit/action@v3.0.1"
```

Comment:
Note we're using pre-commit here to run the various linters. This is something I'd like to eventually get all of the repos doing, though it might take a bit to get there.